win/pkgs/env/toolchain.ps1

```diff
@@ -169,8 +169,39 @@ if (-Not $Env:ROASTER_TOOLCHAIN_COMMITED)
 
     if (${Env:VSCMD_VER} -eq $null)
     {
-        ${VS_HOME} = & "${Env:ProgramFiles(x86)}/Microsoft Visual Studio/Installer/vswhere.exe" -latest -property installationPath
-        Invoke-Expression $($(cmd /c "`"${VS_HOME}/VC/Auxiliary/Build/vcvarsall.bat`" x64 10.0.16299.0 & set") -Match '^.+=' -Replace '^','${Env:' -Replace '=','}="' -Replace '$','"' | Out-String)
+        ${VS_HOME} = & "${Env:ProgramFiles(x86)}/Microsoft Visual Studio/Installer/vswhere.exe" `
+                        -latest -products * `
+                        -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 `
+                        -property installationPath
+
+        $vcvars_script = "${Env:VS_HOME}/VC/Auxiliary/Build/vcvarsall.bat"
+        if (-Not (Test-Path $vcvars_script))
+        {
+            Write-Host "Unable to locate Visual Studio command file: vcvarsall.bat. This is required for VC env import."
+            Exit 1
+        }
+
+        Invoke-Expression $($(cmd /c "`"${Env:VS_HOME}/VC/Auxiliary/Build/vcvarsall.bat`" x64 10.0.16299.0 & set") -Match '^.+=' -Replace '^','${Env:' -Replace '=','}="' -Replace '$','"' | Out-String)
+
+        if ((${Env:VCToolsVersion} -eq $null) -or (${Env:WindowsSDKVersion} -eq $null))
+        {
+            Write-Host "Failed to import VC env."
+            Exit 1
+        }
+
+        if (-not ${Env:VCToolsVersion}.StartsWith("14.1"))
+        {
+            # MSVC Internal version numbering
+            # https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B
```

Comment:
I would probably use `+` instead of `%2B` to make it human readable, since most browsers support that.